kani-compiler/src/kani_compiler.rs

```diff
@@ -121,18 +129,17 @@ enum CompilationStage {
         target_harnesses: Vec<HarnessId>,
         next_harnesses: Vec<Vec<HarnessId>>,
         all_harnesses: HashMap<HarnessId, HarnessInfo>,
+        crate_info: CrateInfo,
+    },
+    Done {
+        metadata: Option<(KaniMetadata, CrateInfo)>,
     },
-    Done,
 }
 
 impl CompilationStage {
     pub fn is_init(&self) -> bool {
         matches!(self, CompilationStage::Init)
     }
-
-    pub fn is_done(&self) -> bool {
-        matches!(self, CompilationStage::Done)
-    }
```

Comment:
Do we even use `is_init` too? I'd remove it.

Reply:
We do

Reply:
I kept `is_init` for now. It is currently used in two different `if` conditions. Let me know if you have any strong reason for us to remove it at this point.